src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        add(value,size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (size == 0) {
+            Node<T> temporaryHead = head;
+            Node<T> firstNodeToAdd = new Node<>(null, value, temporaryHead);
+            tail = firstNodeToAdd;
+            head = firstNodeToAdd;
+            size++;
+        } else if (size > 0 && index == 0) {
+            Node<T> newHead = new Node<>(null, value, head);
+            newHead.next.prev = newHead;
+            head = newHead;
+            size++;
+        } else if (size > 0 && index == size) {
+            Node<T> newTail = new Node<T>(tail, value, null);
+            newTail.prev.next = newTail;
+            tail = newTail;
+            size++;
+        } else {
```

Comment:
too complicated construction. we need to check if size = 0, but try to think this way: can we find nodes from both sides of inserted one? what can we do with them?